docs/conflicts.rst

```diff
@@ -0,0 +1,131 @@
+
+Observe too that the "parents" of a particular Snapshot are a commitment to the state visible by a particular client when creating any new Snapshot.
+
+
+Detecting Conflicts
+-------------------
+
+A new update is communicated to us -- that is, we've downloaded a previously-unknown Snapshot from some other Participant's mutable Capbility.
+
+We now look at our own Snapshot for the corresponding file.
+Either our Snapshot appears in some parents (including grandparents, etc) of the new Snapshot, or it doesn't.
+
+If it does appear, this is an "update" (and we simply replace the local content with the incoming content).
+
+Instead if our Snapshot does not appear as any ancestor of the incoming Snapshot, a conflict is determined.
+
+This is because when the other device created their Snapshot, they didn't know about ours (or else it would appear as an ancestor) so we have made a change "at the same time".
+
+Note that unlike tools like Git, we do not examine the contents of the file or try to produce differences -- everything is determined from the metadata in the Snapshot.
+This means that even if we happened to make the very same edits "at the same time" it would still be a conflict.
+
+
+Showing Conflicts Locally
+-------------------------
+
+Once a conflict is detected, "conflict marker" files are put into the local magic folder location (our local file remains unmodified, and something like ``foo.conflict-desktop`` will appear.
+The state database is also updated (conflicts can also be listed via the API and CLI).
+
+Although magic-folder itself doesn't try to examine the contents, you can now use any ``diff`` or similar tools you prefer to look at what is different between your copy and other participant(s) copies.
+
+
+Resolving Conflicts
+-------------------
+
+We cannot "magically" solve a conflict: two devices produced new Snapshots while not communicating with each other.
+
+Thus, it is up to the humans using these devices to determine what happens.
+Once appropriate changes are decided upon, a new Snapshot is produced with *two or more parents*: one parent for each of the Snapshots involved in the conflict (we've only talked about one other participant so far, but there could be more).
+
+Such a Snapshot (with two or more parents) indicates to the other clients a particular resoltion to the conflict has been decided.
+
+So there's actually another case when we see an incoming new Snapshot: it may in fact *resolve an existing* conflict.
+If this is the case, conflict markers are removed and the local database is updated (i.e. removing the conflict).
+
+It is a human problem if this resolution is not to your particular liking; you can produce an edit again or talk to the human who runs the other computer(s) involved.
+The history of these changes *is available* in the parent (or grandparent) Snapshots if the UX you're using can view or restore these.
+
+
+Resolving via Filesystem
+------------------------
+
+Not currently possible.
```

Comment:
This seems important. A lot of other folder-sync tools expect/require users to resolve conflicts via the filesystem (e.g., by renaming one of the local/marked copies to the "real" filename and deleting the other marked copies) and so users might be naturally inclined to attempt to resolve conflicts this way. Is there another good way/place to emphasize that this won't work for users who might not read/see this document?

Beyond that, are there any plans to support this method of conflict resolution? "Not _currently_ possible" might imply that it could be added later but there's no indication of importance, priority, etc. If there's a ticket relating to this, consider linking it here.

Reply:
#725 mentions it, but there's not a separate ticket.

The note isn't meant to mean this _isn't_ important, but there's not even a great description of what it should do, and this PR is already well into "probably too big" territory so I certainly don't want to add a whole bunch more to it.